src/iotivity/iotivity_instance.cc

```diff
@@ -0,0 +1,152 @@
+/****************************************************************************
+**
+** Copyright © 1992-2014 Cisco and/or its affiliates. All rights reserved.
+** All rights reserved.
+**
+** $CISCO_BEGIN_LICENSE:APACHE$
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+** http://www.apache.org/licenses/LICENSE-2.0
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+** $CISCO_END_LICENSE$
+**
+****************************************************************************/
+#include <string>
+#include <map>
+
+#include "iotivity/iotivity_instance.h"
+#include "iotivity/iotivity_device.h"
+#include "iotivity/iotivity_server.h"
+#include "iotivity/iotivity_client.h"
+#include "iotivity/iotivity_resource.h"
+
+std::map<int, OCRepresentation> ResourcesMap;
+
+IotivityInstance::IotivityInstance() {
+  pDebugEnv = getenv("IOTIVITY_DEBUG");
+  m_device = new IotivityDevice(this, NULL);
+}
+
+IotivityInstance::~IotivityInstance() {
+  delete m_device;
+}
+
+void IotivityInstance::HandleMessage(const char* message) {
+  std::string resp = PrepareMessage(message);
```

Comment:
copy here the content of PrepareMessage and get rid of PrepareMessage.
Later, if ever, you add synchronous comms, you can refactor then.